core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -173,7 +183,23 @@ public synchronized void connect() throws IOException, ConnectionFailedException
           new TBinaryProtocol(mTransportProvider.getClientTransport(mAddress));
       mProtocol = new TMultiplexedProtocol(binaryProtocol, getServiceName());
       try {
-        mProtocol.getTransport().open();
+        // Set a timeout for transport.open() in case it hangs when the authentication type does not
+        // match between Alluxio client and servers.
+        FutureTask<Void> openTask = new FutureTask<>(
+            new TransportOpenCallable(mProtocol.getTransport()));
+        executor.submit(openTask);
+        try {
+          openTask.get((long) socketTimeoutMs, TimeUnit.MILLISECONDS);
+        } catch (TimeoutException e) {
+          String message = "Timeout to open the thrift transport. This exception may be caused by "
+              + "incorrect network configuration. Please check whether the authentication types "
+              + "match between client and server. Note that NOSASL client is not able to connect "
+              + "to servers with SIMPLE security mode.";
+          throw new IOException(message, e);
+        } finally {
+          openTask.cancel(true);
```

Comment:
hmm, does this cancel work? AFAIK, future.cancel only works if the underlying task can be cancelled (for example, sleep).